routes/taskRoutes.js

```diff
@@ -4,12 +4,13 @@ const taskController = require("../controllers/taskController");
 
 const Router = express.Router();
 
-Router.use(checkAuth);
+// Router.use(checkAuth);
 
 Router.post("/", taskController.addTask);
 Router.get("/", taskController.getAllTasks);
 Router.get("/:taskID", taskController.getTask);
 Router.patch("/:taskID", taskController.updateTask);
+Router.put("/:taskID", taskController.markCompleted);
```

Comment:
use put when you want to replace the whole document and patch in case when you want to update specific fields, so mention post type according to the type of operation you are performing on task. in case you want to distinguish between update-task and mark-as-completed please use route path like
-update-task
-mark-as-completed

and also don't forget to update the request type according to the type of operation you are performing on the task